source/slang/slang-ir-explicit-global-context.cpp

```diff
@@ -222,21 +222,14 @@ struct IntroduceExplicitGlobalContextPass
                     // skip any global-scope parameters that are varying instead of
                     // uniform.
                     //
-                    switch (m_target)
+                    if (m_target == CodeGenTarget::CUDASource || isCPUTarget(m_target))
```

Comment:
This seems to miss the `CUDAHeader` case. Please add an `isCUDATarget()` query (if there isn't one) and use that instead.

I tend to prefer a `switch` over an `if` for the precise reason that they are easier to extend with more cases, but I can see how this one would grow unwieldy if you had to list all the LLVM CPU targets as well.